File: _3_algorithms/_1_algo_design/_1_recursion_and_backtracking/_3_fiobanancci.cpp
// C++ code to implement Fibonacci series
#include <iostream>
#include <chrono>
using namespace std;
using namespace std::chrono;

// Function for fibonacci

int fib(int n)
{
    // base condition
    if (n == 0)
        return 0;

    // Self work
    if (n == 1 || n == 2)
        return 1;

    // Recursion function
    else
        return (fib(n - 1) + fib(n - 2));
}

// Driver Code
int main()
{

    // Get starting timepoint
    auto start = high_resolution_clock::now();

    // for loop to print the fiboancci series.
    for (int i = 0; i < 45; i++)
    {
        cout << fib(i) << " ";
    }

    // Get ending timepoint
    auto stop = high_resolution_clock::now();

    auto duration = duration_cast<microseconds>(stop - start);

    cout << "Time taken by function: "
         << duration.count() << " microseconds" << endl;

    return 0;
}

File: _3_b_algorithms_in_python/_1_recursion_and_backtracking/_1_basics/_3_fiobanancci.py
from datetime import datetime

def fib(n):
#base condition
    if n == 0:
        return 0

#Self work
    if n == 1 or n == 2:
        return 1

#Recursion function
    else:
        return fib(n - 1) + fib(n - 2)

#Driver Code
if __name__ == "__main__":
#Get starting timepoint
    start = datetime.now()

#for loop to print the Fibonacci series.
    for i in range(45):
        print(fib(i), end=" ")

#Get ending timepoint
    stop = datetime.now()

    duration = stop - start

    print("\nTime taken by function:", duration)
